explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala

```diff
@@ -0,0 +1,189 @@
+  private val RAColumnId            = ColumnId("ra")
+  private val DecColumnId           = ColumnId("dec")
+  private val TimingWindowsColumnId = ColumnId("timing_windows")
+  private val SEDColumnId           = ColumnId("sed")
+  private val ChargedTimeColumnId   = ColumnId("charged_time")
+
+  private val columnNames: Map[ColumnId, String] = Map(
+    // Default columns
+    GroupsColumnId          -> "Groups",
+    ObservationIdColumnId   -> "Observation Id",
+    ValidationCheckColumnId -> " ",
+    StatusColumnId          -> "Status",
+    CompletionColumnId      -> "Completion",
+    TargetTypeColumnId      -> " ",
+    TargetColumnId          -> "Target",
+    ConstraintsColumnId     -> "Constraints",
+    FindingChartColumnId    -> "Finding Chart",
+    ConfigurationColumnId   -> "Configuration",
+    DurationColumnId        -> "Duration",
+
+    // Default hidden columns
+    PriorityColumnId      -> "Priority",
+    RAColumnId            -> "RA",
+    DecColumnId           -> "Dec",
+    TimingWindowsColumnId -> "Timing Windows",
+    SEDColumnId           -> "SED",
+    ChargedTimeColumnId   -> "ChargedTime"
+  )
+
+  private val DefaultColVisibility: ColumnVisibility = ColumnVisibility(
+    PriorityColumnId      -> Visibility.Hidden,
+    RAColumnId            -> Visibility.Hidden,
+    DecColumnId           -> Visibility.Hidden,
+    TimingWindowsColumnId -> Visibility.Hidden,
+    SEDColumnId           -> Visibility.Hidden,
+    ChargedTimeColumnId   -> Visibility.Hidden
+  )
+
+  private val component = ScalaFnComponent
+    .withHooks[Props]
+    .useContext(AppContext.ctx)
+    // Columns
+    .useMemoBy((_, _) => ())((props, ctx) =>
+      _ =>
+        def column[V](id: ColumnId, accessor: ObsSummaryWithTitleConstraintsAndConf => V)
+          : ColumnDef.Single[ObsSummaryWithTitleConstraintsAndConf, V] =
+          ColDef(id, accessor, columnNames(id))
+
+        List(
+          // TODO: GroupsColumnId
+          column(ObservationIdColumnId, _.id).sortable,
+
+          // TODO: ValidationCheckColumnId
+          column(StatusColumnId, _.status),
+          // TODO: CompletionColumnId
+          // TODO: TargetTypeColumnId
+          column(TargetTypeColumnId, _ => ())
+            .setCell(_ => Icons.Star.withFixedWidth())
+            .setSize(35.toPx),
+          column(TargetColumnId, _.title),
+          column(ConstraintsColumnId, _.constraints).setCell(_.value.summaryString),
+          // TODO: FindingChartColumnId
+          column(ConfigurationColumnId, _.conf),
+          column(DurationColumnId, _.executionTime.toHoursMinutes)
+
+          // TODO: PriorityColumnId
+          // TODO: RAColumnId
+          // TODO: DecColumnId
+          // TODO: TimingColumnId
+          // TODO: SEDColumnId
+          // TODO: ChargedTimeColumnId
+        )
+    )
+    // Rows
+    .useMemoBy((props, _, _) => props.observations.get)((_, _, _) => _.toList)
+    // TODO: useReactTableWithStateStoreBy
```

Comment:
Do we need to add the key to the DB in order to store the prefs?

Reply:
Yes